### PR TITLE
Store standard redeemscript

### PR DESCRIPTION
**Pull request by nathanieliov**

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java [hunk 1]
```diff
@@ -1295,6 +1295,358 @@ void testTxIsProcessableInLegacyVersion() {
...
     }
 
+    @Test
+    void testIsMigrationTx_sending_funds_from_retired_p2sh_fed_to_active_p2sh_fed() {
+        NetworkParameters networkParameters = bridgeConstantsMainnet.getBtcParams();
+        Context btcContext = new Context(networkParameters);
+        when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
+
+        List<BtcECKey> retiredFederationKeys = Stream.of(
+            BtcECKey.fromPrivate(Hex.decode("fc01")),
```
Comment (jeremy-then) [inline rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1305]: Isn't it better to actually use a random hex string with the length of an actual private key? I don't think 'fc01' is a good example of a private key here.

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java [hunk 2]
```diff
@@ -1295,6 +1295,358 @@ void testTxIsProcessableInLegacyVersion() {
...
+            1L,
+            bridgeConstantsMainnet.getBtcParams(),
+            bridgeConstantsMainnet.getErpFedPubKeysList(),
+            bridgeConstantsMainnet.getErpFedActivationDelay(),
+            activations
+        );
+
+        List<BtcECKey> activeFederationKeys = Stream.of(
+            BtcECKey.fromPrivate(Hex.decode("fa01")),
+            BtcECKey.fromPrivate(Hex.decode("fa02"))
```
Comment (jeremy-then) [inline rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1321]: The same here, an actual private key-like hex would be better. Or is this some kind of a seed?

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java [hunk 3]
```diff
@@ -1295,6 +1295,358 @@ void testTxIsProcessableInLegacyVersion() {
...
```
Comment (jeremy-then) [inline rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1391]: Same as above.

**Review by jeremy-then — APPROVED** [review]
LGTM.

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java (hunk 1 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1305]: I would say yes if this test were related to testing something with signing/or with signature. But, this is not the case. It's just testing that for a given btc transaction sending funds to the fed standardP2shScript of each type of Federation, the tx is processed correctly by the bridge.

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java (hunk 2 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1321]: Same answered.

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java [hunk 4]
```diff
@@ -1295,6 +1295,358 @@ void testTxIsProcessableInLegacyVersion() {
...
+        TransactionInput migrationTxInput = new TransactionInput(
+            networkParameters,
+            migrationTx,
+            new byte[]{},
+            new TransactionOutPoint(networkParameters, 0, Sha256Hash.ZERO_HASH)
+        );
+        migrationTx.addInput(migrationTxInput);
+        signWithNecessaryKeys(retiredFederation, retiredFederationKeys, migrationTxInput, migrationTx);
+
+        assertTrue(BridgeUtils.isMigrationTx(
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1474]: We should also assert that it works with the P2SH script, not just the standard one

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java [hunk 5]
```diff
@@ -1295,6 +1295,358 @@ void testTxIsProcessableInLegacyVersion() {
...
+            new TransactionOutPoint(networkParameters, 0, Sha256Hash.ZERO_HASH)
+        );
+        migrationTx.addInput(migrationTxInput);
+        signWithNecessaryKeys(retiringFederation, retiringFederationKeys, migrationTxInput, migrationTx);
+
+        assertTrue(BridgeUtils.isMigrationTx(
+            migrationTx,
+            activeFederation,
+            null,
+            retiringFederation.getStandardP2SHScript(),
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1535]:
```suggestion
            retiringFederation,
            null,
```
You should be passing the retiring federation here, not the p2sh script

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java [hunk 6]
```diff
@@ -1295,6 +1295,358 @@ void testTxIsProcessableInLegacyVersion() {
...
+            new TransactionOutPoint(networkParameters, 0, Sha256Hash.ZERO_HASH)
+        );
+        migrationTx.addInput(migrationTxInput);
+        signWithNecessaryKeys(retiringFederation, retiringFederationKeys, migrationTxInput, migrationTx);
+
+        assertTrue(BridgeUtils.isMigrationTx(
+            migrationTx,
+            activeFederation,
+            retiringFederation,
+            null,
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1643]: Same as above

File: rskj-core/src/test/java/org/ethereum/config/blockchain/upgrades/ActivationConfigsForTest.java [hunk 7]
```diff
@@ -160,6 +160,17 @@ private static List<ConsensusRule> getHop401Rskips() {
         return rskips;
     }
 
+    private static List<ConsensusRule> getFingerroot500Rskips() {
+        List<ConsensusRule> rskips = new ArrayList<>();
+        rskips.addAll(Arrays.asList(
+            ConsensusRule.RSKIP374,
+            ConsensusRule.RSKIP375,
+            ConsensusRule.RSKIP377
```
Comment (marcos-iov) [inline rskj-core/src/test/java/org/ethereum/config/blockchain/upgrades/ActivationConfigsForTest.java:168]: Are those all the RSKIPs we have in fingerroot?

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 8]
```diff
@@ -1403,6 +1396,54 @@ void test_change_powpeg_from_p2shErpFederation_with_mainnet_powpeg_post_RSKIP_35
         );
     }
 
+    @Test
+    void test_change_powpeg_from_p2shErpFederation_with_mainnet_powpeg_post_RSKIP_377_stores_standard_redeemScript() throws Exception {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:1400]: Seems like we are lacking an assertion that the standard p2sh redeem script was stored as expected.

Perhaps getting the value from the storage should be enough to validate it.

We should also check that pre-fingerrot, the non-standard p2sh script is stored. Perhaps this last assertion can be added as part of the other tests, so it's not necessary to create a new test just for this

**Review by marcos-iov — CHANGES_REQUESTED** [review]
Nice job, left a few comments regarding the tests

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java (hunk 4 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1474]: Done.

File: rskj-core/src/test/java/org/ethereum/config/blockchain/upgrades/ActivationConfigsForTest.java (hunk 7 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/org/ethereum/config/blockchain/upgrades/ActivationConfigsForTest.java:170]: Done

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java (hunk 8 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java:1425]: Done

File: rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java (hunk 6 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/BridgeUtilsTest.java:1653]: That is correct, my bad

File: rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java [hunk 9]
```diff
@@ -543,6 +545,29 @@ private void testChangePowpeg(
             false,
             true
         );
+
+        if (oldPowPegFederationType == FederationType.erp || newPowPegFederationType == FederationType.erp){
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java]: Changed

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**